File: Backend/Model/blog.types.go
package model

import "go.mongodb.org/mongo-driver/bson/primitive"

type Reactions struct {
	Likes    int `json:"likes"`
	Dislikes int `json:"dislikes"`
}

type Blog struct {
	ID        primitive.ObjectID `bson:"_id,omitempty" json:"id"`
	Title     string             `json:"title"`
	Body      string             `json:"body"`
	Reactions Reactions          `json:"reactions"`
	Tags      []string           `json:"tags"`
	Views     int                `json:"views"`
	UserId    int                `json:"userid"`
}

File: src/app/Models/Blog.types.ts
export interface Post {
    id: number;
    title: string;
    body: string;
    tags: string[];
    reactions: {
      likes: number;
      dislikes: number;
    };
    views: number;
    userId: number;
}

export interface newPost {
  title: string;
  body: string;
  tags: string[];
  reactions: {
    likes: number;
    dislikes: number;
  };
  views: number;
  userId: number;
}

export interface PostsResponse {
    filter(arg0: (blog: any) => boolean): PostsResponse;
    posts: Post[];
    total:number;
    skip:number;
    limit:number;
}
